autoPyTorch/evaluation/abstract_evaluator.py

```diff
@@ -447,7 +550,39 @@ def finish_up(self, loss: Dict[str, float], train_loss: Dict[str, float],
         * saving the files for the ensembles_statistics
         * generate output for SMAC
         We use it as the signal handler so we can recycle the code for the
-        normal usecase and when the runsolver kills us here :)"""
+        normal usecase and when the runsolver kills us here :)
+
+        Args:
+            loss (Dict[str, float]):
+                The optimization loss, calculated on the validation set. This will
+                be the cost used in SMAC
+            train_loss (Dict[str, float]):
+                The train loss, calculated on the train set
+            opt_pred (np.ndarray):
+                The predictions on the validation set. This validation set is created
+                from the resampling strategy
+            valid_pred (Optional[np.ndarray]):
+                Predictions on a user provided validation set
+            test_pred (Optional[np.ndarray]):
```

Comment:
Probably, we should discuss the usage of test and validation sets given by user.
Because we currently do not use test dataset at all although it is provided.
Also, since user splits dataset and the dataset that is created before user splits will be stored on RAM.
It also increases the memory usage.

Reply:
We do use the test! We score the test performance to closely track the overfit. If no test data is provided, then no consumption here.